src/App/DocumentObjectPyImp.cpp

```diff
@@ -75,23 +75,39 @@ Py::Object DocumentObjectPy::getDocument() const
     }
 }
 
-PyObject*  DocumentObjectPy::addProperty(PyObject *args)
+PyObject*  DocumentObjectPy::addProperty(PyObject *args, PyObject *kwd)
 {
     char *sType,*sName=nullptr,*sGroup=nullptr,*sDoc=nullptr;
     short attr=0;
     std::string sDocStr;
     PyObject *ro = Py_False, *hd = Py_False;
-    if (!PyArg_ParseTuple(args, "s|ssethO!O!", &sType,&sName,&sGroup,"utf-8",&sDoc,&attr,
-        &PyBool_Type, &ro, &PyBool_Type, &hd))
+    PyObject* enumVals = nullptr;
+    static char *kwlist[] = {"type","name","group","doc","attr","read_only","hidden","enum_vals",nullptr};

```

Comment:
This should now be changed to
```suggestion
    const std::array<const char *, 9> kwlist {"type","name","group","doc","attr","read_only","hidden","enum_vals",nullptr};
```

and the call to `PyArg_ParseTupleAndKeywords` replaced with a call to `Wrapped_ParseTupleAndKeywords` -- this is made for C++ compliance, and is required for Qt 6 on MSVC.

Reply:
Will do. Did you also make these changes after merging #10269? Also, specifying the `9` seems inelegant.

Reply:
I am taking care of all the ones in the current codebase, so you don't need to worry about those. And yes, it would be nice if the compiler would supply the 9 for us, but it cannot: although the **number** of elements must be constant in a `std::array`, their actual contents is not required to be, and you can default-initialize it "empty" which will default-init all of its elements.

Reply:
Just pushed the changes, though I cannot confirm right now on my local machine if it compiles.